File: Study Plan/Algorithm I/Longest_Substring_without_Repeating.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        seen = set()
        result = 0
        start = 0
        for i in range(len(s)):
            while s[i] in seen:
                seen.remove(s[start])
                start += 1
            seen.add(s[i])
            result = max(result, i - start + 1)
        return result
        

File: Study Plan/Algorithm I/Longest_Substring_without_Repeating.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int i = 0, j, N = s.size(), result = 0;
        int A[128] = {0}; 
        for(j = 0; j < N; j ++){
            i = max(i, A[s[j]]);
            result = max(result, j - i + 1);
            A[s[j]] = j + 1;
        }
        return result;
    }
};